internal/services/cognitive/cognitive_account_customer_managed_key_resource.go

```diff
@@ -0,0 +1,224 @@
+		Pending:    []string{"Accepted"},
+		Target:     []string{"Succeeded"},
+		Refresh:    cognitiveAccountStateRefreshFunc(ctx, client, *id),
+		MinTimeout: 15 * time.Second,
+		Timeout:    time.Until(timeout),
+	}
+
+	if _, err = stateConf.WaitForStateContext(ctx); err != nil {
+		return fmt.Errorf("waiting for update of %s: %+v", id, err)
+	}
+	d.SetId(id.ID())
+
+	return resourceCognitiveAccountCustomerManagedKeyRead(d, meta)
+}
+
+func resourceCognitiveAccountCustomerManagedKeyRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Cognitive.AccountsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.AccountID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		return fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+	if resp.Properties == nil || resp.Properties.Encryption == nil {
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("cognitive_account_id", id.ID())
+	if props := resp.Properties.Encryption.KeyVaultProperties; props != nil {
+		var keyName string
+		if props.KeyName != nil {
+			keyName = *props.KeyName
+		}
+
+		var keyVaultUri string
+		if props.KeyVaultURI != nil {
+			keyVaultUri = *props.KeyVaultURI
+		}
+
+		var keyVersion string
+		if props.KeyVersion != nil {
+			keyVersion = *props.KeyVersion
+		}
+		keyVaultKeyId, err := keyVaultParse.NewNestedItemID(keyVaultUri, "keys", keyName, keyVersion)
+		if err != nil {
+			return fmt.Errorf("parsing `key_vault_key_id`: %+v", err)
+		}
+		d.Set("key_vault_key_id", keyVaultKeyId.ID())
+		if props.IdentityClientID != nil {
+			d.Set("identity_client_id", *props.IdentityClientID)
+		}
+	} else {
+		d.Set("key_vault_key_id", nil)
+		d.Set("identity_client_id", nil)
+	}
+
+	return nil
+}
+
+func resourceCognitiveAccountCustomerManagedKeyDelete(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Cognitive.AccountsClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.AccountID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	locks.ByName(id.Name, "azurerm_cognitive_account")
+	defer locks.UnlockByName(id.Name, "azurerm_cognitive_account")
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		return fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+
+	if resp.Properties == nil || resp.Properties.Encryption == nil {
+		return fmt.Errorf("%s doesn't exist", id)
+	}
+
+	// set key source to Microsoft.CognitiveServices to disable customer managed key
+	props := cognitiveservices.Account{
+		Properties: &cognitiveservices.AccountProperties{
+			Encryption: &cognitiveservices.Encryption{
+				KeySource: cognitiveservices.KeySourceMicrosoftCognitiveServices,
+			},
+		},
+	}
+
+	if _, err = client.Update(ctx, id.ResourceGroup, id.Name, props); err != nil {
+		return fmt.Errorf("updating %s: %+v", id, err)
```

Comment:
this'll be a pointer so this needs to be:

```suggestion
		return fmt.Errorf("removing Customer Managed Key for %s: %+v", *id, err)
```